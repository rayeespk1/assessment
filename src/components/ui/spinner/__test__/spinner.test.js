import React from "react";
import '@testing-library/jest-dom'
import { render, cleanup } from "@testing-library/react";
import Spinner from "../spinner";

afterEach(cleanup);

test("renders", () => {
    const { asFragment } = render(<Spinner />);
    expect(asFragment()).toMatchSnapshot();
});

test("loading correct data", () => {
    const { getByTestId, getByText } = render(<Spinner />);

    expect(getByTestId("loaderTag")).toHaveTextContent("Loading...");
    expect(getByText("Loading...")).toHaveClass("Loader");
});