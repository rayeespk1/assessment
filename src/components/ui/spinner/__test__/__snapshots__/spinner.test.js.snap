// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="Loader"
    data-testid="loaderTag"
  >
    Loading...
  </div>
</DocumentFragment>
`;
